### ISSUE TITLE
Docs: search input accessibility improvements

### ISSUE THREAD

**Issue by jbadan** [SAP/fundamental-styles#185]
https://www.w3.org/TR/wai-aria-practices/#combobox

1) Search input for docs page does not have appropriate keyboard controls:
```
Keyboard Interaction
Tab: The textbox is in the page Tab sequence.
Note: The popup indicator icon or button (if present), the popup, and the popup descendants are excluded from the page Tab sequence.
Textbox Keyboard Interaction
When focus is in the textbox:

Down Arrow: If the popup is available, moves focus into the popup:
If the autocomplete behavior automatically selected a suggestion before Down Arrow was pressed, focus is placed on the suggestion following the automatically selected suggestion.
Otherwise, places focus on the first focusable element in the popup.
Up Arrow (Optional): If the popup is available, places focus on the last focusable element in the popup.
Escape: Dismisses the popup if it is visible. Optionally, clears the textbox.
Enter: If an autocomplete suggestion is automatically selected, accepts the suggestion either by placing the input cursor at the end of the accepted value in the textbox or by performing a default action on the value. For example, in a messaging application, the default action may be to add the accepted value to a list of message recipients and then clear the textbox so the user can add another recipient.
Printable Characters: Type characters in the textbox. Note that some implementations may regard certain characters as invalid and prevent their input.
Standard single line text editing keys appropriate for the device platform (see note below).
Alt + Down Arrow (Optional): If the popup is available but not displayed, displays the popup without moving focus.
Alt + Up Arrow (Optional): If the popup is displayed:
If the popup contains focus, returns focus to the textbox.
Closes the popup.
```

See example 1 in this link: https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html

2) Missing roles/aria-attributes

* `input` element: `role='combobox' aria-expanded='true/false', aria-autocomplete='list', aria-owns='results-container', aria-label='Search'`
* Results container `ul` is missing: `role='listbox' aria-label='Search results'`
* `li` missing `role='option'`
* `a` missing `tabIndex='-1'`

**Comment by MattL75** [comment]
Anybody up for removing the search input completely and just having a combobox instead? Lots of redundant stuff here.

**Comment by saad-mo** [comment]
search-input is used in shell bar as well which is different styling than Combobox - https://sap.github.io/fundamental-styles/demo-pages/full-width-app-demo.html

**Comment by MattL75** [comment]
Yeah we're just going to use a different skin for combobox in the shellbar. No point in maintaining two components that are so similar. Especially search-input which isn't even on ui5.

**Comment by jbadan** [comment]
@saad-mo @MattL75 Sorry for any confusion here, I was _only_ referring to the docs site search tool. While our search input component needs some changes, this is a different implementation (at least for fundamental-styles). 

This should be any easy 5 minute fix whenever anyone has time to take it on.

<img width="650" alt="Screen Shot 2019-07-22 at 7 58 38 PM" src="https://user-images.githubusercontent.com/29607818/61653776-5bc4d980-acbb-11e9-84b8-9ce839ec26ae.png">

**Comment by jbadan** [comment]
Closing per move to storybook
